packages/terra-modal/src/Modal.jsx

```diff
@@ -79,8 +83,28 @@ const defaultProps = {
   role: 'document',
 };
 
-/* eslint-disable react/prefer-stateless-function */
+const KEYCODES = {
+  ESCAPE: 27,
+};
+
 class Modal extends React.Component {
+  constructor() {
+    super();
+    this.handleKeydown = this.handleKeydown.bind(this);
+  }
+
+  componentDidMount() {
+    if (this.props.closeOnEsc) {
```

Comment:
If, for whatever reason, someone passed the closeOnEsc prop to the Modal component after its initial mounting, we wouldn't add the event handler.

I'm not sure how realistic a scenario that is, but we could always add the event handler and check the prop in the  handler to cover that case.